dataprofiler/profilers/profile_builder.py

```diff
@@ -1223,17 +1231,74 @@ def _min_col_samples_used(self):
         i.e. every column in the Profile was read up to this row (possibly
         further in some cols)
         """
-        return min([self._profile[col].sample_size
-                    for col in self._profile], default=0)
+        return min([col.sample_size for col in self._profile], default=0)
 
     @property
     def _min_sampled_from_batch(self):
         """
         Calculates and returns the number of rows that were completely sampled
         in the most previous batch
         """
-        return min([self._profile[col]._last_batch_size
-                    for col in self._profile], default=0)
+        return min([col._last_batch_size for col in self._profile], default=0)
+
+    @property
+    def is_initialized(self):
+        """
+        Determines if profiler has been initialized with data
+        """
+        return self.total_samples > 0
+
+    @staticmethod
+    def _get_and_validate_schema_mapping(schema1, schema2, strict=False):
+        """
+        Validate compatibility between schema1 and schema2 and return a dict
+        mapping indices in schema1 to their corresponding indices in schema2.
+        In __add__: want to map self _profile idx -> other _profile idx
+        In _update_profile_from_chunk: want to map data idx -> _profile idx
+
+        :param schema1: a column name to index mapping
+        :type schema1: Dict[str, list[int]]
+        :param schema2: a column name to index mapping
+        :type schema2: Dict[str, list[int]]
+        :param strict: whether or not to strictly match (__add__ case)
+        :type strict: bool
+        :return: a mapping of indices in schema1 to indices in schema2
+        :rtype: Dict[int, int]
+        """
+
+        if strict and (len(schema1) == 0 or len(schema2) == 0):
+            raise ValueError("Cannot merge empty profiles.")
+
+        # In the case of _update_from_chunk with uninitialized schema
+        if not strict and len(schema2) == 0:
+            return {col_ind: col_ind for col_ind_list in schema1.values()
+                    for col_ind in col_ind_list}
```

Comment:
could assign lens at beginning to var so don't do multiple times

Reply:
Implemented